public/static/app/perfdiscovery/PerfDiscoveryDataService.js

```diff
@@ -433,8 +450,8 @@ mciModule.factory('PerfDiscoveryDataService', function (
   // Processes `queryData` return value
   // Populates dicts of current (now, baseline) and history result items
   // returns dict of {now, baseline, history} test result items
-  function processData(promise) {
-    return promise.then(onProcessData)
+  function processData(promise, metricName) {
+    return promise.then(onProcessData(metricName))
```

Comment:
Isn't this wrong? __onProcessData(metricName)__ executes immediately before the promise returns. 

Do you actually want:

```
return promise.then(() => onProcessData(metricName))
```

Reply:
I think this is okay since onProcessData returns a function

Reply:
I changed it to return a function, though it's still named confusingly. I changed the name so it's clearer hopefully, but if you have a suggestion on how to better pass a parameter to a delegate I can do that